okhttp/src/main/java/okhttp3/CertificatePinner.kt

```diff
@@ -90,13 +90,19 @@ import javax.net.ssl.SSLPeerUnverifiedException
  *
  * Wildcard pattern rules:
  *
- *  1. Asterisk `*` is only permitted in the left-most domain name label and must be the only
- *     character in that label (i.e., must match the whole left-most label). For example,
- *     `*.example.com` is permitted, while `*a.example.com`, `a*.example.com`, `a*b.example.com`,
- *     `a.*.example.com` are not permitted.
- *  2. Asterisk `*` cannot match across domain name labels. For example, `*.example.com` matches
+ *  1. Where used, asterisk `*` (or double-asterisk `**`) must be the only text in the label. For
+ *     example, `*.example.com` is permitted, while `*a.example.com`, `a*.example.com`, and
+ *     `a*b.example.com` are not permitted.
+ *  2. Asterisk `*` can appear multiple times but must not appear in the top-level label. For
```

Comment:
Is anyone asking for this? I’d really just prefer three modes

`example.com`
`*.example.com`
`**.example.com`

The multiple asterixes seems like an unnecessary departure from the SubjectAltName feature that we’re trying to emulate here.

Reply:
Personally I don't plan on using multiple asterisks, but I can conceive of uses for it. If you feel like it wouldn't be beneficial then I'm happy to get rid of it.